@xen-orchestra/lite/src/composables/collection-filter.composable.ts

```diff
@@ -1,25 +1,27 @@
+import { getFirst } from "@/libs/utils";
 import * as CM from "complex-matcher";
 import { computed, ref, watch } from "vue";
 import { type LocationQueryValue, useRoute, useRouter } from "vue-router";
 
 interface Config {
   queryStringParam?: string;
+  initialFilters?: string[];
 }
 
-export default function useCollectionFilter(
-  config: Config = { queryStringParam: "filter" }
-) {
+export default function useCollectionFilter<T>(config: Config = {}) {
   const route = useRoute();
   const router = useRouter();
-  const filtersSet = ref(
-    config.queryStringParam
-      ? queryToSet(route.query[config.queryStringParam] as LocationQueryValue)
-      : new Set<string>()
-  );
+  const { queryStringParam, initialFilters = [] } = config;
+  const filtersSet = ref<Set<string>>(new Set(initialFilters));
   const filters = computed(() => Array.from(filtersSet.value.values()));
 
-  if (config.queryStringParam) {
-    const queryStringParam = config.queryStringParam;
+  if (queryStringParam !== undefined) {
+    const queryString = route.query[queryStringParam];
+
+    if (queryString !== undefined) {
+      filtersSet.value = queryToSet(getFirst(queryString));
+    }
```

Comment:
In this case:
```ts
useCollectionFilter({
  initialFilters: ["foo:bar"],
  queryStringParam: "filters",
});
```
This does not work because `watch(filters, () => ...)` won't be triggered since we don't use `addFilter()` to update the `filters` computed.
I think you need to add to the `watch` options: `immediate: true`

Reply:
I'm not sure to understand.

`watch(filter...)` is meant to update the URL when adding/removing a filter.

If we don't call `addFilter()`... then it's expected that the watch callback is not called.

Reply:
Ping @MathieuRA

Reply:
Maybe I don't quite understand what `initialFilters` should do.
When I use `initialFilters`, I expect this filter to be present when the page loads.

Reply:
`initialFilters` are applied by default when no filters are passed in the URL.

Let's take your example:
```ts
useCollectionFilter({
  initialFilters: ["foo:bar"],
  queryStringParam: "filters",
});
```

In this case, the `foo:bar` filter will be applied if there is no `?filters=...` in the URL.

If later you call `addFilter("name:/^foo/")`, then the URL will be updated thanks to the `watch`.